usd_qtpy/render_util/dialog.py

```diff
@@ -0,0 +1,889 @@
+        self.chk_purpose_proxy.setChecked(True)
+        self.chk_purpose_guide.setChecked(False)
+
+        purpose_vlayout = QtWidgets.QVBoxLayout()
+        purpose_vlayout.addWidget(self.chk_purpose_default)
+        purpose_vlayout.addWidget(self.chk_purpose_render)
+        purpose_vlayout.addWidget(self.chk_purpose_proxy)
+        purpose_vlayout.addWidget(self.chk_purpose_guide)
+
+        self.formlayout.addRow("Included purposes:",purpose_vlayout)
+
+        # Complexity combobox
+        self.cbox_complexity = QtWidgets.QComboBox()
+        self.cbox_complexity.addItems(playblast.get_complexity_levels())
+        self.cbox_complexity.setCurrentIndex(2)
+        self.formlayout.addRow("Complexity / Quality",self.cbox_complexity)
+
+        # Renderer Combobox
+        self.cbox_renderer = QtWidgets.QComboBox()
+        self.cbox_renderer.addItems(playblast.iter_renderplugin_names())
+        self.cbox_renderer.setCurrentIndex(0)
+        self.formlayout.addRow("Renderer",self.cbox_renderer)
+        
+        self.vlayout.addLayout(self.formlayout)
+
+        # Ui post hook
+        self.ui_post_hook(self.vlayout)
+
+        self.vlayout.addStretch()
+
+        # Playblast button
+
+        self.btn_playblast = QtWidgets.QPushButton()
+        self.btn_playblast.setText("Playblast!")
+        self.vlayout.addWidget(self.btn_playblast)
+
+        self.btn_playblast.clicked.connect(self.playblast_callback)
+
+        # Progress bar
+        self.progressbar = QtWidgets.QProgressBar(self)
+        self.progressbar.setFormat("Not started...")
+        self.vlayout.addWidget(self.progressbar)
+
+    def _update_fit(self):
+        if "New: Framing Camera" in self.cbox_camera.currentText():
+            self.spinbox_fit.setDisabled(False)
+            self.lbl_fit.setDisabled(False)
+        else:
+            self.spinbox_fit.setDisabled(True)
+            self.lbl_fit.setDisabled(True)
+
+    def _update_resolution(self, res_tuple: tuple[int,int]):
+        self.spinbox_horresolution.setValue(res_tuple[0])
+        self.spinbox_verresolution.setValue(res_tuple[1])
+
+    def _update_framerange_options(self):
+        index = self.cbox_framerange_options.currentIndex()
+        if index == 0:
+            self.spinbox_frame_start.setDisabled(False)
+            self.spinbox_frame_end.setDisabled(True)
+            self.spinbox_frame_stride.setDisabled(True)
+        elif index == 1:
+            self.spinbox_frame_start.setDisabled(False)
+            self.spinbox_frame_end.setDisabled(False)
+            self.spinbox_frame_stride.setDisabled(False)
+        elif index == 2:
+            self.spinbox_frame_start.setDisabled(True)
+            self.spinbox_frame_end.setDisabled(True)
+            self.spinbox_frame_stride.setDisabled(True)
+
+    def _prompt_renderoutput(self):
+        filename = prompt_output_path("Render result to...")
+        if filename:
+            self.txt_filename.setText(os.path.normpath(filename))
```

Comment:
If currently a path is set in the text field it should actually default to that path when browsing again.

Reply:
Will implement, after checking path validity of course.